iree/tools/run_lit.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+import re
+import subprocess
+from typing import List, Tuple, Union
+
+
+def parse_arguments():
+  parser = argparse.ArgumentParser(description="run_lit.py helper.")
+  parser.add_argument("--test_file", help="Path of the test file to execute.",
+                      required=True)
+  args = parser.parse_args()
+  return args
+
+
+def execute_in_shell(command: str,
+                     shell: str = '/bin/bash') -> Tuple[Union[str, int]]:
+  """Executes command and returns its stdout, stderr and returncode."""
+  p = subprocess.Popen([shell, command], shell=True, stdout=subprocess.PIPE,
+                       stderr=subprocess.PIPE)
+  stdout, stderr = p.communicate()
+  return stdout, stderr, p.returncode
+
+
+def find_runfiles(runfiles_dir: str) -> List[str]:
+  runfiles = []
+  # Iterate through all files in runfiles_dir and its subdirectories.
+  for (path, _, files) in os.walk(runfiles_dir):
+    for file_name in files:
+      file_path = os.path.join(path, file_name)
+
+      # Check each file to see if the user can execute it.
+      if os.access(file_path, os.X_OK):
+        runfiles.append(file_path)
+  return runfiles
+
+
+def get_runline_commands(test_file: str) -> List[str]:
+  """Returns all of the commands listed in the given test file."""
+  with open(test_file, 'r') as f:
+    lines = f.readlines()
+  commands = []
+  for i, line in enumerate(lines):
+    match = re.match('^// RUN: *', line)
+    if match:
+      commands.append(line[match.span()[-1]:])
+      if not commands[-1]:
+        print(f'ERROR: Encountered an empty runline on line {i + 1} of '
+              f'{test_file}')
+        exit(1)
```

Comment:
Honestly a bit confused by the block. Took me a couple reads (and looking up match.span) to grok it. Since you're just using regex for prefix matching, I think it shouldn't be necessary at all.

What about:

```suggestion
    if line.startswith(RUNLINE_PREFIX):
      command = line[len(RUNLINE_PREFIX):].strip()
      if not command:
        print(f'ERROR: Encountered an empty runline on line {i + 1} of '
              f'{test_file}')
        exit(1)
      commands.append(command)
```